### ISSUE TITLE
A11y: provide content skip links for keyboard/screen reader users

### ISSUE THREAD

**Issue by RyKilleen** [forem/forem#1153]
**Is your feature request related to a problem? Please describe.**
When users navigate a page with keyboard tabbing, screen readers, or other assistive technology, a useful pattern is to provide a "Skip to Main Content' skip link before navigation or long lists (Tags list, etc).

**Describe the solution you'd like**
A visible skip link that pops up when a user has tabbed into an element like the nav or long tags list.

**Additional context**
https://webaim.org/techniques/skipnav/

**Comment by Link2Twenty** [comment]
This is part of the ARIA spec too 
**2.4.1 Bypass Blocks:** A mechanism is available to bypass blocks of content that are repeated on multiple Web pages. (Level A)

**Useful links:**
https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html
https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=241#qr-navigation-mechanisms-skip

**Comment by jessleenyc** [comment]
Thanks for the feature request. This is approved if anyone wants to jump on it!

**Comment by Link2Twenty** [comment]
Just to add to this, I had to press the tab button 80 times before I could go to the first post.

**Comment by Link2Twenty** [comment]
Adding this to just before the dev logo in the nav

```html
<a href="#articles-list" class="skip-content">Skip to content</a>
```

and this to your CSS

```CSS
skip-content {
  position: absolute;
  top: 0;
  left: 50%;
  transform: translatex(-50%);
  background: black;
  color: white;
  z-index: 9999;
  padding: 12px 8px;
  cursor: default;
  pointer-events: none;
  opacity: 0;
}

skip-content:focus {
  pointer-events: auto;
  opacity: 1;
}
```
**Normal view:**
![image](https://user-images.githubusercontent.com/3534427/52788149-f871b400-3057-11e9-98ac-a7d3f46d3d81.png)
**Focused view:**
![image](https://user-images.githubusercontent.com/3534427/52788182-13dcbf00-3058-11e9-8a93-99350f9fb1fe.png)

Which is not something the everyday user would notice. But would make life, hopefully, a lot easier for keyboard users.

Thoughts?

**Comment by RyKilleen** [comment]
@Link2Twenty I think this demonstrates a great, straight-forward implementation!

**Comment by RyKilleen** [comment]
Tackling this currently!

**Comment by maestromac** [comment]
This is partially implemented am I correct?

**Comment by RyKilleen** [comment]
Correct, right now it's a pretty naive implementation that only works on the home page.

**Comment by scottaohara** [comment]
Could there not be a unique ID that was added to the `h1` of each post, so that the skip link could navigate people there?

Was rather excited to see there was a skip link on the site, and then was immediately disappointed to realize it didn't do anything on anyone's posts. :(

**Comment by RyKilleen** [comment]
Another way to solve it is to use the `<main>` element across templates to indicate where the skip link should jump to. The benefits are that it's more semantic, doesn't require a global magic ID, and any template should be able to identify the main content.

**Comment by RyKilleen** [comment]
The most semantic route forward is probably to guarantee that an `<main id="main-content">` tag exists around the content you want skipped to for any given template. That would keep it semantic and consistent without having to insert magic elements in specific spots for all templates.

**Comment by scottaohara** [comment]
"the most semantic route forward" is a debatable statement,  but doing a quick audit of the site, a fair one based on the lack of proper heading structure / general misuse of heading for a good portion of the site that aren't people's posts.

That said, it seems that a `main` element w/`id` would also need to be properly added to the templates.

**Comment by colabottles** [comment]
@nickytonline I'll take a look at some point with this. Thanks!

**Comment by techieeliot** [comment]
Hey, @colabottles I'm working on issue #4807 but I'd love to help you out over a live stream sometime to finish this PR up.

**Comment by colabottles** [comment]
> Hey, @colabottles I'm working on issue #4807 but I'd love to help you out over a live stream sometime to finish this PR up.

Appreciate the offer @techieeliot  I have no time to stream coming up with a huge project being shipped out in the next week or so. I have one issue left that I need to go back and look at, so perhaps at some point, if you want to schedule a quick Zoom call I can fit that in, the first half of next week (Thursday/Friday I'll be Out Of Office) https://calendly.com/toddlibby/

I have yet to fully look at the remaining issue @nickytonline posted but will be doing so at some point later today in-between meetings.

@nickytonline Appreciate you taking a look at the remaining issue and I am hoping to sort this out, even when Hacktoberfest ends. My schedule has been hectic but I'm always around if you have questions.

**Comment by nickytonline** [comment]
Whoever takes this on, create small PRs. We can do one page at a time.

**Comment by aitchiss** [comment]
This issue is closed since we now have added the skip link functionality to all pages 🎉

There is a small piece of work remaining - currently when a user navigates internally (link click), the skip link isn't the first item focused on the new page. This is captured in #13505, should be a quick fix and once implemented users should be able to get the most out of the new skip links 😄
